### PR TITLE
Appcues test, round 2

### PR DESCRIPTION
**Pull request by orangejenny**
New users in the test will have a template app created for them and be redirected to web apps, where they'll see an appcues flow guiding them through the app.

Review by commit.

@emord / @millerdev 
or @jmtroth0

### CODE REVIEW

File: corehq/apps/app_manager/static/app_manager/json/template_apps/appcues.json [hunk 1]
```diff
@@ -0,0 +1,1191 @@
+{
```
Comment (jmtroth0) [inline corehq/apps/app_manager/static/app_manager/json/template_apps/appcues.json]: Out of curiosity, how did you generate this?

File: corehq/apps/app_manager/views/apps.py [hunk 2]
```diff
@@ -392,6 +396,20 @@ def _inner(request, link_domain, data, master_domain=domain):
...
+def app_from_template(request, domain, slug):
+    send_hubspot_form(HUBSPOT_APP_TEMPLATE_FORM_ID, request)
+    clear_app_cache(request, domain)
+    template = load_app_template(slug)
+    app = import_app_util(template, domain, {
+        'created_from_template': '%s' % slug,
+    })
+    comment = _("A sample application you can try out in Web Apps")
+    build = make_async_build(app, request.user.username, release=True, comment=comment)
+    cloudcare_state = '{{"appId":"{}"}}'.format(build._id)
```
Comment (jmtroth0) [inline corehq/apps/app_manager/views/apps.py]: Do the curly braces get encoded in the url?

**Review by jmtroth0 — APPROVED** [review]
lgtm, just a couple q's

File: corehq/apps/analytics/ab_tests.py [hunk 3]
```diff
@@ -23,23 +24,22 @@ def __init__(self, config, request):
...
-        if self._cookie_id not in self.request.COOKIES:
-            version = random.sample(self.options, 1)[0]  # todo weighted options
-        else:
-            version = self.request.COOKIES[self._cookie_id]
-        return version
+    def version(self, assign_if_blank=True):
+        if self._cookie_id in self.request.COOKIES:
+            return self.request.COOKIES[self._cookie_id]
+        if assign_if_blank:
+            return random.sample(self.options, 1)[0]
```
Comment (millerdev) [inline corehq/apps/analytics/ab_tests.py]: Can be more succinctly phrased as `random.choice(self.options)`

**Review by millerdev — APPROVED** [review]

File: corehq/apps/app_manager/static/app_manager/json/template_apps/appcues.json (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/app_manager/static/app_manager/json/template_apps/appcues.json]: https://www.commcarehq.org/a/template-apps/apps/download/ae67c8f583012d45d1270e458a93036c/ - Amelia & Sheel built this app ages ago when we first released a template apps feature.

File: corehq/apps/app_manager/views/apps.py (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/app_manager/views/apps.py]: I don't think so

**Review by millerdev — APPROVED** [review]

**Comment by orangejenny** [comment]
Planning to release Monday - could I get a quick sanity check on the last commit?

File: corehq/apps/app_manager/views/apps.py [hunk 4]
```diff
@@ -404,6 +407,21 @@ def app_from_template(request, domain, slug):
...
+    for path in get_template_app_multimedia_paths(slug):
+        media_class = None
+        with open(os.path.join(app_template_dir(slug), path), "rb") as f:
+            f.seek(0)
+            data = f.read()
+            media_class = CommCareMultimedia.get_class_by_data(data)
+            if media_class:
+                multimedia = media_class.get_by_data(data)
+                multimedia.attach_data(data,
+                                       original_filename=path.split(os.sep)[-1],
```
Comment (millerdev) [inline corehq/apps/app_manager/views/apps.py]: nit: there is a function for that: `os.path.basename(path)`

**Review by millerdev — APPROVED** [review]
Do you know what the size (in bytes) of all these multimedia files are? If it's large (more than a few kb), would you consider using smaller media files to not grow the size of our repo unnecessarily?

**Comment by orangejenny** [comment]
@millerdev It's fairly large, 1 or 2 MB total. There are a bunch of audio files. Is your concern about how long it'll take to create the template app?

**Comment by orangejenny** [comment]
@millerdev Sorry, I missed that you said "to not grow the size of our repo unnecessarily". I'm inclined to defer judgement until we find out if this test is successful.

File: corehq/apps/app_manager/views/apps.py (hunk 4 above)
Comment (orangejenny) [reply corehq/apps/app_manager/views/apps.py]: thanks, fixed

**Comment by millerdev** [comment]
> I'm inclined to defer judgement until we find out if this test is successful.

That's fine. In order to have those large files not be part of the git history you'll have to rebase and edit (amend) commit https://github.com/dimagi/commcare-hq/pull/21225/commits/fc814c027ef7e1a012185e84ed0a0ecf040a437e, replacing each of the media files. I imagine the content is not important, so you could just replace the content of each of those files with a few random bytes.

**Comment by orangejenny** [comment]
@millerdev thanks, I hadn't thought about amending the commits afterward.

**Comment by millerdev** [comment]
I assume you decided against making the attachments smaller?

**Comment by orangejenny** [comment]
Yeah - given that there's audio and the test is meant to be a realistic app, I don't think there's a good way to get them down to a small size. I think if the test is successful we (Marshall & I) should talk about trimming down the multimedia.

**Comment by jmtroth0** [comment]
It might be worth pulling the assets from S3 or some alternative in the future if this is a concern

**Pull request merged**